.github/workflows/tests.yml

```
@@ -72,4 +72,9 @@ jobs:
 
     - name: Tests
       run: |
-        pytest tests --ignore tests/integration_tests
+        if [ ${{ matrix.python-version }} = 3.9 ]; then
+          pytest tests --ignore tests/integration_tests \
+            --ignore tests/samplers_tests/test_samplers.py
```

Only two test functions matter for python 3.9, so how about using `--deselect` to exclude the two test functions?

```suggestion
            --deselect tests/samplers_tests/test_samplers.py::test_nan_objective_value --deselect tests/samplers_tests/test_samplers.py::test_partial_fixed_sampling
```

Ref: https://docs.pytest.org/en/stable/example/pythoncollection.html#deselect-tests-during-test-collection

---

Thanks for the review. Seems reasonable 👍